workitem/enum_type.go

```diff
@@ -56,6 +56,24 @@ func (t EnumType) Equal(u convert.Equaler) bool {
 	return true
 }
 
+// checkValueSetIsEnclosing checks if a given set is a subset of a superset.
+func (t EnumType) checkValueSetIsEnclosing(subset []interface{}, superset []interface{}) bool {
+	set := make(map[interface{}]int)
```

Comment:
Go has the concept of an empty struct (`struct{}{}`). It consumes no memory and is the preferred choice for this kind of task. This implies that you can no longer do the counting but that's not needed here IMHO. Please see this example: https://play.golang.org/p/imrLNYTpUBM . Please note, that I am iterating over the subset first while you do it reversed. 

```go
// checkValueSetIsEnclosing checks if a given set is a subset of a superset.
func (t EnumType) checkValueSetIsEnclosing(subset []interface{}, superset []interface{}) bool {
	toBeFound := map[interface{}]struct{}{}
	for _, v := range subset {
		toBeFound[v] = struct{}{}
	}
	for _, s := range superset {
		delete(toBeFound, s)
	}
	return len(toBeFound) == 0
}
```